### DISCUSSION TITLE
Discord earner

### DISCUSSION THREAD

**Discussion by newtarkprojects** [gameserverapp/Platform#422] [Suggestions]
Hey Max!

It would be cool if the discord bot was able to award players token amounts for chatting in certain channels or look for certain responses. 

If at all possible, have chat games that also award tokens for example a quiz, first to answer correctly gets 100 shop tokens.

Thank you,
Gem

**Comment by mvaessen** [comment]
Hi Gem,

Thanks for your suggestion!
I posted this message to provide little bit of background info about potential implications:

While your suggestion is technically possible, this is currently a bit limited with regards to how Discord is handling privacy lately. 

Not too long ago Discord made changes to their API's and their privacy policy about how companies like GSA can access data. This is mainly to protect people privacy, which I think was a great move. As you can imagine can it be a problem if any bot is reading what you're saying and passing this to who knows where. So to combat that Discord restricted access to certain privileges (reading messages & user presence) and allowed this on a case-by-case basis where companies like GSA had to explain in detail why they needed certain privileges and also how they deal with storing that data or not.
If Discord thought it was a good addition to their platform companies would get permissions and if not; good luck.

GSA had to get a special exemption for the cluster chat + discord chat feature too. And I'm really happy we got that.

Changing how GSA processes chat or the purpose (new features) would mean that GSA has to re-apply and get an exemption for this new case, which is not guaranteed at all. This is apart from the development hours it takes to implement such a feature, which will take time for sure.

Realistically is this not something I'm planning on added in the near future, mostly due to the amount of time it takes. Regardless I do think it's a great suggestion and please keep 'em coming ;)

- Max

**Comment by newtarkprojects** [comment]
Hey Max,

Right, it's understandable they have that measure in place.

Would a widget be classed as the same thing? For example the bot posts a widget, first reaction or first button pressed on the widget would award the tokens? 

Thank you,
Gem

**Reply by mvaessen** [reply]
I'm not sure how exactly widgets work to be honest. Never looked into that too much. Based on the docs it seems to be limited to message content only atm, so I think that would be possible. See this article for more info: https://support-dev.discord.com/hc/en-us/articles/4404772028055-Message-Content-Privileged-Intent-FAQ

Anything involving reading a message does fall under the privacy rules for sure.

**Comment by mvaessen** [comment]
rejecting over earlier mentioned limitations
